tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -3616,29 +3642,39 @@ void MSMain(uint GID : SV_GroupIndex,
     #else // NO FALLBACK
 
     #define ROOT_SIG [RootSignature("RootFlags(CBV_SRV_UAV_HEAP_DIRECTLY_INDEXED | SAMPLER_HEAP_DIRECTLY_INDEXED | ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT)")]
+    
+    static ConstantBuffer<Constants> CB = ResourceDescriptorHeap[12];
 
-    static ConstantBuffer<Constants> CB = ResourceDescriptorHeap[6];
-
-    static ByteAddressBuffer g_rawBuf = ResourceDescriptorHeap[CB.idx0];
-    static StructuredBuffer<float> g_structBuf = ResourceDescriptorHeap[CB.idx1];
-    static Texture2D<float> g_tex = ResourceDescriptorHeap[CB.idx2];
-    static RWByteAddressBuffer g_rwRawBuf = ResourceDescriptorHeap[CB.idx3];
-    static RWStructuredBuffer<float> g_rwStructBuf = ResourceDescriptorHeap[CB.idx4];
-    static RWTexture1D<float> g_rwTex = ResourceDescriptorHeap[CB.idx5];
+    #define NON_UNIFORM
+    #ifndef NON_UNIFORM
+    #define INDEXER(ix) ix
+    #define OFFSET(ix) 0
+    #else
+    #define INDEXER(ix) NonUniformResourceIndex(ix)
+    #define OFFSET(ix) ix % 2
+    #endif
 
     // Result buffers
-    static RWStructuredBuffer<float> g_result = ResourceDescriptorHeap[7];
-    static RWStructuredBuffer<float> g_resultVS = ResourceDescriptorHeap[8];
-    static RWStructuredBuffer<float> g_resultPS = ResourceDescriptorHeap[9];
+    static RWStructuredBuffer<float> g_result = ResourceDescriptorHeap[13];
+    static RWStructuredBuffer<float> g_resultVS = ResourceDescriptorHeap[14];
+    static RWStructuredBuffer<float> g_resultPS = ResourceDescriptorHeap[15];
     
     static SamplerState g_samp = SamplerDescriptorHeap[CB.idx0];
     static SamplerComparisonState g_sampCmp = SamplerDescriptorHeap[CB.idx1];
 
     #endif // FALLBACK
 
     ROOT_SIG
-    [NumThreads(1, 1, 1)]
+    [NumThreads(8, 8, 1)]
     void main(uint ix : SV_GroupIndex) {
+    
+        ByteAddressBuffer g_rawBuf = ResourceDescriptorHeap[INDEXER(CB.idx0) + OFFSET(ix)];
```

Comment:
Non-uniform resource indexes must use the `NonUniformResourceIndex()` intrinsic around the index inside the indexing operation, like:
```
ByteAddressBuffer g_rawBuf = ResourceDescriptorHeap[NonUniformResourceIndex(INDEXER(CB.idx0) + OFFSET(ix))];
```